sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/SequencePointCollector.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.dotnet.tests;
+
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SequencePointCollector {
+  private static final Logger LOG = Loggers.get(SequencePointCollector.class);
+  private Set<SequencePoint> sequencePoints = new HashSet<>();
+
+  public void add(SequencePoint sequencePoint){
+    // we only want to count covered sequences when these are on the same line
+    if (sequencePoint.getStartLine() == sequencePoint.getLineEnd()) {
+      sequencePoints.add(sequencePoint);
+    }
+  }
+
+  public void publishCoverage(Coverage coverage){
+    Map<String, List<SequencePoint>> sequencePointsPerFile = sequencePoints.stream().collect(Collectors.groupingBy(SequencePoint::getFilePath));
```

Comment:
I think it will be more readable if you iterate in a functional manner with `stream()` - to avoid the long definitions which are not bringing value

Reply:
Sorry, but I don't understand what you mean :) can you give an example?

Reply:
this method body can consist only of
```
sequencePoints.stream()
  .collect(Collectors.groupingBy(SequencePoint::getFilePath))
  forEach(fileSequencePoints -> publishCoverage(coverage, fileSequencePoints.getKey(), fileSequencePoints.getValue()));
```

Reply:
good idea. done.